packages/@jsii/kernel/src/kernel.ts

```diff
@@ -15,6 +15,28 @@ import * as onExit from './on-exit';
 import * as wire from './serialization';
 import * as tar from './tar-cache';
 
+export enum JsiiErrorType {
+  JSII_FAULT = '@jsii/kernel.Fault',
+  JS_ERROR = '@jsii/kernel.JSError',
```

Comment:
While I think `JavascriptError`/`JavascriptException` "makes sense" in other languages, I think this isn't quite descriptive of the semantics in the TypeScript/JS code... Literally every exception being handled here is a Javascript error...

So while I think it may be okay to use `JavascriptException` in Java (I mean we could bike shed here - do we *need* to surface the fact this was JS? Maybe... maybe not?), the JavaScript code should refer to this as a something more along the lines of `@jsii/kernel.RuntimeError` or maybe even `@jsii/kernel.ForwardedRuntimeError`.

The enum should likely also have JSDoc on it...